### ISSUE TITLE
Dragging changed files from a commit to the workspace gives an error

### ISSUE THREAD

**Issue by daniilS** [gitbutlerapp/gitbutler#4317]
### Version

0.5.596

### Operating System

Windows

### Distribution Method

msi (Windows)

### Describe the issue

When you drag a file from a commit, you can drag it to the workspace of other virtual branches (but not to to the workspace on the current branch, surprisingly). However, doing that gives the following error:

_**Failed to update hunk ownership**
Invalid args `branch` for command `update_virtual_branch`: ownership ranges cannot be empty_

Either dragging a committed file to a workspace should uncommit the changes, or it shouldn't be possible to drag the file there.

In addition, dragging a file onto the commit it originally came from gives a _**Failed to amend commit** no rebase was performed_ error, when it should probably do nothing at all.

![image](https://github.com/gitbutlerapp/gitbutler/assets/6144189/565aa3a0-6105-4a3b-887c-780be356131d)

**Comment by PavelLaptev** [comment]
Yep, having the same error. It doesn't appear to be related to the UI.

<img width="487" alt="image" src="https://github.com/user-attachments/assets/e02c5e3e-b3db-4c3b-9ca6-20f99a73ceae">

https://github.com/user-attachments/assets/f965c782-a73f-41dc-9eaa-951eb075812d

Additionally, I've noticed that if I drag the file from a commit, the commit shows the `Amend commit` state. This shouldn't happen because we can't amend a file in the same commit.

https://github.com/user-attachments/assets/020513d4-97bc-493e-ab72-0d2b3cd4bfb0

Probably @Caleb-T-Owens could help here

**Comment by eproxus** [comment]
This also happens if one drags a changed file _into_ a previously committed commit. Strangely enough if one drags it several times it finally works without the error and the file gets amended into the commit.

**Comment by krlvi** [comment]
This has since been fixed in the current app. Dragging committed changes back to the workspace now uncommits them via https://github.com/gitbutlerapp/gitbutler/pull/8551, shipped in [0.14.22](https://github.com/gitbutlerapp/gitbutler/releases/tag/release/0.14.22). Dropping a file or hunk back onto the commit it already belongs to is blocked via https://github.com/gitbutlerapp/gitbutler/pull/8510, shipped in [0.14.20](https://github.com/gitbutlerapp/gitbutler/releases/tag/release/0.14.20).

Sorry this sat around for so long. Closing as completed.
